samples/bluetooth/df_per_adv_beacon/src/main.c

```diff
@@ -0,0 +1,208 @@
+
+struct bt_df_adv_cte_tx_param cte_params = {
+	.cte_len = CTE_LEN,
+	.cte_count = 1,
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX)
+	.cte_type = BT_HCI_LE_AOD_CTE_2US,
+	.num_ant_ids = ARRAY_SIZE(ant_patterns),
+	.ant_ids = ant_patterns
+#else
+	.cte_type = BT_HCI_LE_AOA_CTE,
+	.num_ant_ids = 0,
+	.ant_ids = NULL
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_TX */
+	};
+
+static void adv_sent_cb(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_sent_info *info)
+{
+	printk("Advertiser[%d] %p sent %d\n", bt_le_ext_adv_get_index(adv),
+	       adv, info->num_sent);
+}
+
+static struct bt_le_ext_adv *advertising_set_create(void)
+{
+	struct bt_le_ext_adv *adv;
+	int err;
+
+	printk("Advertising set create...");
+	err = bt_le_ext_adv_create(&param, &adv_callbacks, &adv);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return NULL;
+	}
+	printk("success\n");
+	return adv;
+}
+
+static int periodic_advertising_params_set(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	printk("Periodic advertising params set...");
+	err = bt_le_per_adv_set_param(adv, &per_adv_param);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	return 0;
+}
+
+static int periodic_advertising_enable(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	printk("Periodic advertising enable...");
+	err = bt_le_per_adv_start(adv);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	printk("Extended advertising enable...");
+	err = bt_le_ext_adv_start(adv, &ext_adv_start_param);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	char addr_s[BT_ADDR_LE_STR_LEN];
+	bt_addr_le_t addr = {0};
+	size_t count = 1;
+
+	bt_id_get(&addr, &count);
+	bt_addr_le_to_str(&addr, addr_s, sizeof(addr_s));
+
+	printk("Started extended advertising as %s\n", addr_s);
+
+	return 0;
+}
+
+static int cte_params_set(struct bt_le_ext_adv *adv,
+			  struct bt_df_adv_cte_tx_param *params)
+{
+	int err;
+
+	printk("Update CTE params...");
+	err = bt_df_set_adv_cte_tx_param(adv, params);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	return 0;
+}
+
+static int cte_enable(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	printk("Enable CTE...");
+	err = bt_df_adv_cte_tx_enable(adv);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	return 0;
+}
+
+static void bt_ready(int err)
+{
+	printk("Bluetooth initialization...\n");
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success\n");
+
+	adv_set = advertising_set_create();
+	if (!adv_set) {
+		return;
+	}
+
+	err = cte_params_set(adv_set, &cte_params);
+	if (err) {
+		return;
+	}
+
+	err = periodic_advertising_params_set(adv_set);
+	if (err) {
+		return;
+	}
+
+	err = cte_enable(adv_set);
+	if (err) {
+		return;
+	}
+
+	err = periodic_advertising_enable(adv_set);
+	if (err) {
+		return;
+	}
```

Comment:
No need of second level of abstraction in the sample application, users should see the direct use of Bluetooth Host API use. (You are conveying that the API signatures are not good enough, needing abstractions).

Reply:
It is a kind of leftover. I used the code in a more complicated scenarios, so I wanted to avoid multiple printk() with debug information. I'll remove 2nd level functions.